app/models/concerns/with_usages.rb

```diff
@@ -16,6 +16,21 @@ def usage_in_organization_of_type(type, organization = Organization.current)
     item.is_a?(type) ? item : nil
   end
 
+  class_methods do
+    def content_aggregate_of(association)
+      aggregate_of association
```

Comment:
Perhaps instead of introducing a new method `content_aggregate_of`, you could redefinie `agregate_of` in this mixing to add the desired behavior. That way you don't need two very similar methods, and allow using `agregate_of` without the indexig logic.

 
I think this should look like the following, thanks to mixin linearization: 

```ruby
def aggregate_of(association)
      super
     ..your code...
```

Reply:
yeah, that makes way more sense! :+1:

Reply:
actually, rethinking this, it would still be pretty obnoxious that way. I think method would end up looking something like this:
```ruby
def aggregate_of(association)
  super

  revamp :rebuild! do |_, this, children, hyper|
    old_children = this.send association
    added_children = children - old_children
    hyper.(children)
    this.usages.each { |it| it.index_children(added_children) }

    self
  end
end
```

Reply:
Nice!